src/client.rs

```diff
@@ -1,32 +1,35 @@
 use std::fmt;
 use std::sync::Arc;
 
-use crate::http::Method;
-use crate::Request;
+use crate::middleware::{Middleware, Next};
+use crate::{HttpClient, Request, Response, Result};
 
-use http_client::HttpClient;
+use futures::future::BoxFuture;
 
 #[cfg(all(feature = "native-client", not(feature = "h1-client")))]
 use http_client::native::NativeClient;
 
 #[cfg(feature = "h1-client")]
 use http_client::h1::H1Client;
 
-/// An HTTP client, capable of creating new `Request`s.
+/// An HTTP client, capable of sending `Request`s and running a middleware stack.
 ///
 /// # Examples
 ///
 /// ```no_run
 /// # #[async_std::main]
 /// # async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
 /// let client = surf::Client::new();
-/// let req1 = client.get("https://httpbin.org/get").recv_string();
-/// let req2 = client.get("https://httpbin.org/get").recv_string();
-/// let (str1, str2) = futures::future::try_join(req1, req2).await?;
+/// let res1 = client.recv_string(surf::get("https://httpbin.org/get"));
+/// let res2 = client.recv_string(surf::get("https://httpbin.org/get"));
+/// let (str1, str2) = futures::future::try_join(res1, res2).await?;
```

Comment:
I think it would be really nice if we could replicate the `surf::*` top level methods on the client, such that this example could be written as

```rust
let res1 = client.get("https://httpbin.org/get").recv_string();
let res2 = client.get("https://httpbin.org/get").recv_string();
```

Reply:
This unfortunately doesn't make much sense...? What would `client.get()` return? A single `Request` stored on a reusable `Client` surely isn't it. A `RequestBuilder`? That would mean having it store an `Option<Client>` too...

Reply:
I still disagree with it but this has been done in 8977cba49059e1762b191855ac9fba4bb845a0c4, @dignifiedquire please take a look again if you are able to.